packages/x-date-pickers-pro/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -64,20 +70,29 @@ const useUtilityClasses = (ownerState: OwnerState) => {
     isPreviewing,
     isStartOfPreviewing,
     isEndOfPreviewing,
+    isFirstVisibleCell,
+    isLastVisibleCell,
+    isHiddenDayFiller,
     selected,
     classes,
   } = ownerState;
 
   const slots = {
     root: [
       'root',
-      isHighlighting && !outsideCurrentMonth && 'rangeIntervalDayHighlight',
-      (isStartOfHighlighting || isStartOfMonth) && 'rangeIntervalDayHighlightStart',
-      (isEndOfHighlighting || isEndOfMonth) && 'rangeIntervalDayHighlightEnd',
+      isHighlighting && 'rangeIntervalDayHighlight',
+      isStartOfHighlighting && 'rangeIntervalDayHighlightStart',
+      isEndOfHighlighting && 'rangeIntervalDayHighlightEnd',
+      outsideCurrentMonth && 'rangeIntervalOutsideCurrentMonth',
```

Comment:
> but it doesn't make sense having this class when outside of current month dates are not in the selected range and when overriding it looks a bit nasty in case of the following style override

Yes, I used `rangeInterval` like a prefix for all of root classes. I did not realised it was a prefix for days inside the selected range

Reply:
Hm... That's a good point. 🤔 
Maybe I went too quickly to this conclusion...
After looking at the prefixes (class names) we currently have, it sort of makes sense to have it, because the selected days have `DayHighlight` naming.
But maybe that's a call for us to re-evaluate these class names consistency and correctness..? 🤔

Reply:
I just removed the `rangeInterval` prefix when it was not meaning full